### ISSUE TITLE
[BUG] Error on existing subscription when trying to add another subscription

### ISSUE THREAD

**Issue by cerpass-86LAK** [SAP/terraform-provider-btp#344]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What version of Terraform are you using?

0.2.0-beta

### What type of issue are you facing

bug report

### Describe the bug

![image](https://github.com/SAP/terraform-provider-btp/assets/124213814/cb216c48-9f82-41ea-9317-5316bd6b81f7)

throws the above error when trying to add another subscription after already creating a subscription in an earlier apply

### Expected Behavior

no errors. 

### Steps To Reproduce

Setup and create the infrastructure and include the below block. 
```hcl
resource "btp_subaccount_subscription" "businessApplicationStudio" {
  subaccount_id = btp_subaccount.subaccount.id
  app_name      = "sapappstudiotrial"
  plan_name     = "trial" 
}
```

Then after applying the above and creating the infrastructure. Add the below block and try applying the changes. The below block will force the existing subscription to fail with the above shown api error. 
```hcl
resource "btp_subaccount_subscription" "launchpadService" {
  subaccount_id = btp_subaccount.subaccount.id
  app_name      = "SAPLaunchpad"
  plan_name     = "standard" 
}
```

### Add screenshots to help explain your problem

The planned changes for BAS subscription is shown below even though it should not be impacted
![image](https://github.com/SAP/terraform-provider-btp/assets/124213814/2aad2b34-0218-46b6-b4b8-3df248195385)

**Comment by lechnerc77** [comment]
I am not able to reproduce the error. Here my Setup:

Initial Setup with entitlement and subscription to BAS:

```hcl
###
# Creation of subaccount
###
resource "btp_subaccount" "project" {
  name      = local.project_subaccount_name
  subdomain = local.project_subaccount_domain
  region    = lower(var.region)
}

###
# Entitlement of subaccount for BAS - plan "standard"
###
resource "btp_subaccount_entitlement" "bas_standard" {
  subaccount_id = btp_subaccount.project.id
  service_name  = "sapappstudio"
  plan_name     = "standard-edition"
}

###
# App subscription of subaccount for BAS - plan "standard"
###
resource "btp_subaccount_subscription" "businessApplicationStudio" {
subaccount_id = btp_subaccount.project.id
app_name = "sapappstudio"
plan_name = "standard-edition"
depends_on = [ btp_subaccount_entitlement.bas_standard ]
}
```
Result of first `terraform apply`:

![FirstRun](https://github.com/SAP/terraform-provider-btp/assets/22294087/7c39ddbc-10ff-430d-b382-733327822ea9)

After that I added the following blocks as you described to the `main.tf`:

```hcl
###
# Entitlement of subaccount for SAP Build Workzone Standard Edition - plan "standard"
###
resource "btp_subaccount_entitlement" "saplaunchpad_standard" {
  subaccount_id = btp_subaccount.project.id
  service_name  = "SAPLaunchpad"
  plan_name     = "standard"
}

###
# App subscription for SAP Build Workzone Standard Edition - plan "standard"
###
resource "btp_subaccount_subscription" "launchpadService" {
subaccount_id = btp_subaccount.project.id
app_name = "SAPLaunchpad"
plan_name = "standard"
depends_on = [ btp_subaccount_entitlement.saplaunchpad_standard ]
}

```

Result of second `terraform apply`:

![SecondRun](https://github.com/SAP/terraform-provider-btp/assets/22294087/f65e3eb0-f496-4fdc-b2dc-bd5b8681ac4a)

@cerpass-86LAK Did you perform any steps in between that impacted the Terraform state?

**Comment by lechnerc77** [comment]
@cerpass-86LAK : That sounds like the root cause for the issue and is indeed something we detected as a side effect in #351.
The issue #351 is fixed with PR #352 and will be available in the next release of the Terraform provider.
I will therefore close this issue
